Delete List Using the Delete List Button

ksdenisova opened the pull request:
**As** a user
**I want** to be able to delete active list 
**So that** I can use the common button for this

**Given:** ToDoit app is running
**And:** The active has chosen 
**When:** The user press 'Delete List' button
**Then:** The active list has been deleted

**Acceptance:**
- Remove delete images from list titles

[Trello](https://trello.com/c/ksaL3iYd/16-delete-list-using-the-delete-list-button)

Current design:
<img width="1465" alt="image" src="https://user-images.githubusercontent.com/89826596/155418267-1008b8d5-34a3-4ef1-9f7e-5e119aee99fc.png">

Expected design:
<img width="1399" alt="image" src="https://user-images.githubusercontent.com/89826596/154993994-8aa86117-fbb6-400a-80ad-c044274d72ec.png">

ksdenisova committed:
Rename ToDoApp to ToDoit, add new styles

ksdenisova committed:
Apply a new font, resize boxes

ksdenisova committed:
Fix font rendering, refactor project structure

ksdenisova committed:
Add ListTitles component, refactor ToDoit, add style for list titles

ksdenisova committed:
Refactor displaying lists titles

ksdenisova committed:
Fix margin in body, set up height, rename ToDoit to Home

ksdenisova committed:
Change marker to icon"
"

ksdenisova committed:
Change add image to AddIcon, style textfield for a new item

ksdenisova committed:
Merge branch 'main' into setup-lists-layout

ksdenisova committed:
Rename ListView files, remove unused CSS styles

ksdenisova committed:
Move components to different folders

ksdenisova committed:
Display header for active list

ksdenisova committed:
Change chackbox to mui component, change ListItem style

ksdenisova committed:
Rename ActionForm to NewItem

ksdenisova committed:
Change add image to icon

ksdenisova committed:
Fix items padding

ksdenisova committed:
Change Checkbox icon

ksdenisova committed:
Merge branch 'main' into setup-items-layout

ksdenisova committed:
Create Delete List button

ksdenisova committed:
Style Delete List button

ksdenisova committed:
Delete list using the common button, remove images from list titles

grace000 commented on to-do-app/src/ToDoApp.css:
```diff
@@ -1,28 +1,6 @@
 .Action-Button{
```
I see some mix and matching of css selector naming. You can save the work for an upcoming PR, but I would recommend you lower case the css selectors. Here's a few articles on convention:

https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

https://css-tricks.com/bem-101/

https://dev.to/xavortm/css-naming-conventions-nen

Whichever route you go, try to make sure you're being consistent.

ksdenisova replied:
The `ToDoApp.css` is the original CSS styles we created with Pedro. During the refactoring, I'm removing styles from this file and applying new styles. I'm using lower case for naming selectors in `style.css` files. The final goal is to delete this file completely when no one of those CSS selectors is active. 

Thank you for advice!

grace000 commented on to-do-app/src/components/ListItem/index.js:
```diff
@@ -23,21 +27,22 @@ class ListItem extends Component {
+            disableRipple
+            />
+          <div className="item-text">{this.props.itemProperties.toDoItem}</div>
+            <img src="/remove-item.png"
```
Image tags are self closing 🙂` <img/>`

ksdenisova replied:
I didn't know that. Changed it, thanks!

grace000 commented on to-do-app/src/components/ListView/index.js:
```diff
@@ -22,17 +26,22 @@ class ListView extends Component {
       view = <>
-        <ActionForm activeList = {this.props.activeList} saveActiveList = {this.props.saveActiveList}/>
         {items}
+        <ActionForm activeList = {this.props.activeList} saveActiveList = {this.props.saveActiveList}/>
```
Could you remove the space: `<ActionForm activeList={this.props.activeList} saveActiveList={this.props.saveActiveList}/>`

ksdenisova replied:
Sure, and not only in this place.

grace000 commented on to-do-app/src/components/NewItem/index.js:
```diff
@@ -0,0 +1,45 @@
+                  value={this.state.actionFormValue} placeholder='Enter New To-Do Item'
+            onChange = {event => this.setState({actionFormValue: event.target.value})}>
+          </input>
+        <div className = "List-Item-Button"/>
```
Why not use `<ListItemButton>` or `<Button btnClass={'list-item'} />` ?

ksdenisova replied:
This `div` didn't affect anything. I removed it.

grace000 commented on to-do-app/src/components/ListView/style.css:
```diff
@@ -0,0 +1,23 @@
+  margin-bottom: 46px;
+  margin-left: 38px;
+  width: 183px;
+  font-family: 'Roboto', sans-serif;
```
I think if you set this on the body at the App level, you won't have to add it to all of your selectors: 
```
body {
   font-family: 'Some Font'
}
```

ksdenisova replied:
I set this in main css file `index.css`, but I've noticed that if I need to make font bold or lighter I need to import exact style for [Roboto](https://fonts.google.com/specimen/Roboto?query=roboto) font. This why in some places I specify this again.

grace000 replied:
Hmmm ... not necessary for this PR, but I would try to think of a way to standardize font weight and family so that you don't have to re-implement in your selectors

ksdenisova replied:
Okay, I will. Thanks!

ksdenisova committed:
Remove img closing tag and unused div, rename ActionForm to NewItem

ksdenisova committed:
Remove spaces

ksdenisova committed:
Rename actionFormValue and inputValue

ksdenisova committed:
Merge branch 'main' into delete-list

ksdenisova commented:
@grace000 I removed closing tag for `img`, unused `div` and spaces. I also remembered what Dave told us about naming the `inputValue` and `actionFormValue`, so I renamed them too.
